### DISCUSSION TITLE
User Viewtemplates and Folding Tiddler

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#6717] [General]
Tiddlywiki allows to simply render extra contents using user viewtemplates. But when a tiddler is folded these contents do not follow the folding status. User has to add extra code to view template to check the folding state.

My question/suggestion is: 

Why not user added viewtemplates follow folding state? It seems they should work the same as body content.
Also, these viewtemplate does not use the font size of body content, it seems this is another issue. (see https://github.com/Jermolene/TiddlyWiki5/discussions/6718)

**Comment by pmario** [comment]
>Also, these viewtemplate does not use the font size of body content, it seems this is another issue.

Then it should be a new issue. Otherwise there is a high chance one or the other problem will be lost here.

**Reply by kookma** [reply]
> Then it should be a new issue.

I create a separate issue.

EDITED: See https://github.com/Jermolene/TiddlyWiki5/discussions/6718

**Comment by Jermolene** [comment]
Hi @kookma a user view template selected by the cascade is entirely responsible for rendering a tiddler, and so it has to be responsible for implementing folding. Otherwise the view template couldn't decide which elements of the tiddler should be hidden while folded (we don't hide the entire tiddler, just some view template segments).

It's much the same for the view template body: one of its responsibilities is establishing the rendering environment for the tiddler body because it is different than for the tiddler itself (eg font size).

**Reply by kookma** [reply]
> a user view template selected by the cascade is entirely responsible for rendering a tiddler, and so it has to be responsible for implementing folding. Otherwise the view template couldn't decide which elements of the tiddler should be hidden while folded

Seems fair. If we could know the user view template targets body, then we could expect the same folding behavior as body.
